addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -379,10 +379,14 @@ export const editorCommands = {
                 children.forEach(child => newBlock.appendChild(child));
             }
         }
-        const newRange = new Range();
-        newRange.setStart(startContainer,startOffset);
-        newRange.setEnd(endContainer,endOffset);
-        getDeepRange(editor.editable, { range: newRange, select: true, });
```

Comment:
Don't we want to keep the deep range where we can? Why not keep this logic after your `if`s instead of using `setSelection`?

Reply:
Indeed we should keep the deep range, I have made the changes...